### PR TITLE
Extrahop: Upgraded the app to support for Extrahop cloud, implemented on-poll a…

### PR DESCRIPTION
**Pull request by crestdatasystems**
…ction

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by spopat-crest** [comment]
@crestdatasystems Currently, we are in the process of acquiring the asset of ExtraHop cloud, once done we will prioritize the PR for GA.

**Comment by spopat-crest** [comment]
@crestdatasystems We now have the cloud asset and the development team has started working on the automated failures fixation.

File: extrahop.json [hunk 1]
```diff
@@ -1,51 +1,173 @@
...
+            "data_type": "string",
+            "order": 15
+        },
+        "limit_bytes": {
+            "description": "Maximum number of return bytes for packet retrieval",
+            "data_type": "string",
+            "order": 16
+        },
+        "limit_search_duration": {
+            "description": "Maximum runtime for packet retrieval",
```
Comment (alexa-phantom) [inline extrahop.json:141]: Is this in seconds?

File: extrahop.json [hunk 2]
```diff
@@ -429,13 +553,13 @@
                     "order": 0
                 },
                 "minutes": {
-                    "description": "Minutes to look back (default 30)",
+                    "description": "Amount of lookback data (default is 30 mins)",
```
Comment (alexa-phantom) [inline extrahop.json:556]:
```suggestion
                    "description": "Amount of lookback data in minutes (default is 30 mins)",
```

File: extrahop.json [hunk 3]
```diff
@@ -755,13 +881,13 @@
         {
             "action": "get protocols",
             "identifier": "get_protocols",
-            "description": "Get a list of protocols that a device communicated in the last N minutes",
-            "verbose": "This action retrieves a list of all of the protocols that a device communicated over the last N minutes, optionally filtered by role. Either 'ip' or 'eh_api_id' parameter is required. If both the parameters are provided, 'eh_api_id' will be considered.",
+            "description": "Retrieve a list of protools observed on a device",
```
Comment (alexa-phantom) [inline extrahop.json:884]:
```suggestion
            "description": "Retrieve a list of protocols observed on a device",
```

File: extrahop.json [hunk 4]
```diff
@@ -770,13 +896,13 @@
                     "order": 0
                 },
                 "minutes": {
-                    "description": "Minutes to look back (default 30)",
+                    "description": "Amount of lookback data (default is 30 mins)",
```
Comment (alexa-phantom) [inline extrahop.json:899]:
```suggestion
                    "description": "Amount of lookback data in minutes (default is 30 mins)",
```

File: extrahop.json [hunk 5]
```diff
@@ -883,49 +1009,51 @@
...
+                        "tag"
                     ],
-                    "default": "any",
-                    "order": 0
+                    "default": "custom_filter",
+                    "order": 1
                 },
                 "minutes": {
-                    "description": "Minutes of look back (default 30)",
+                    "description": "Amount of lookback data (default is 30 mins)",
```
Comment (alexa-phantom) [inline extrahop.json:1036]: Fix all similar occurrences

File: extrahop.json [hunk 6]
```diff
@@ -1548,6 +1703,435 @@
...
+                        "Application",
+                        "Device Group",
+                        "Device Group Summary",
+                        "Device",
+                        "Network"
+                    ],
+                    "default": "Application"
+                },
+                "object_id": {
+                    "description": "API IDs for the objects(Comma-delimited)",
```
Comment (alexa-phantom) [inline extrahop.json:1752]:
```suggestion
                    "description": "API IDs for the objects (Comma-delimited)",
```

File: extrahop_consts.py [hunk 7]
```diff
@@ -13,37 +13,105 @@
...
-EXTRAHOP_ERR_MSG_UNAVAILABLE = 'Error message unavailable. Please check the asset configuration and|or action parameters'
+EXTRAHOP_ERROR_MESSAGE_UNAVAILABLE = 'Error message unavailable. Please check the asset configuration and|or action parameters'
 EXTRAHOP_DEBUG_REST_ENDPOINT = "Making REST call to {}"
-EXTRAHOP_STATE_FILE_CORRUPT_ERR = "Error occurred while loading the state file due to its unexpected format. " \
+EXTRAHOP_STATE_FILE_CORRUPT_ERROR = "Error occurred while loading the state file due to its unexpected format. " \
     "Resetting the state file with the default format."
+EXTRAHOP_CONTAINER_LIMIT_ERROR = "Container count should be greater than or equal to the number of object_ids"
+EXTRAHOP_METRIC_REQUIRED_PARAM = "Required parameters: metric_cycle_length, object_type, object_id, metric_category and metric_name"
+EXTRAHOP_METRIC_REQUIRED_API_PARAM = "API required parameters: cycle, object_type, object_ids, metric_category and metric_specs"
+EXTRAHOP_METRIC_NOT_NEEDED_MESSAGE = "For asset_id: {}, Interval should be in multiple of cycle length"
```
Comment (alexa-phantom) [inline extrahop_consts.py:34]:
```suggestion
EXTRAHOP_METRIC_NOT_NEEDED_MESSAGE = "For asset_id: {}, Interval should be in multiples of cycle length"
```
Kind of confusing wording

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: extrahop.json (hunk 1 above)
Comment (aparekh-crest) [reply extrahop.json]: It is in milliseconds by default, we can add other units using unit suffix as well, Added this change

**Review by alexa-phantom — APPROVED** [review]

**Review by ricardo-phantom — APPROVED** [review]

**Pull request merged**